src/main/java/uk/gov/pay/ledger/transaction/dao/TransactionDao.java

```diff
@@ -213,6 +213,18 @@ public class TransactionDao {
                     "agreement_id = EXCLUDED.agreement_id " +
                     "WHERE EXCLUDED.event_count >= transaction.event_count;";
 
+    private static final String REDACT_PII_FROM_TRANSACTIONS =
+            "UPDATE transaction t " +
+                    "set reference = '<REDACTED>'," +
+                    "  cardholder_name = (case when cardholder_name IS NOT NULL then '<REDACTED>' end), " +
+                    "  email = (case when email IS NOT NULL then '<REDACTED>' end), " +
+                    "  transaction_details = JSONB_SET(" +
+                    "                              JSONB_SET(transaction_details, '{address_line1}','\"<REDACTED>\"', false), " +
+                    "                              '{address_line2}','\"<REDACTED>\"', false" +
+                    "                        )" +
+                    "                         -'{reference,cardholder_name,email}'::text[]" +
```

Comment:
Because I'm not familiar with this syntax, what is the purpose of `-'{reference,cardholder_name,email}'::text[]`?

Reply:
This will remove fields from `transaction_details` JSON data.

Mentioned in PR desc

Also mentioned in PR desc

> Removed fields `reference, cardholder_name, email` from transaction_details JSON as these are not used anywhere and are also top-level DB columns. So having them in transaction_details as `<REDACTED>` doesn't add any value

Reply:
Ah, the `-` means "minus these fields".